### DISCUSSION TITLE
how to start transition on initial render?

### DISCUSSION THREAD

**Discussion by ioucyf** [sveltejs/kit#10706] [Q&A]
Hello, 

It says [here](https://svelte.dev/docs/element-directives#transition-fn) in the documentation the following: 
> By default intro transitions will not play on first render. You can modify this behaviour by setting intro: true when you [create a component](https://svelte.dev/docs/client-side-component-api) and marking the transition as global.

I have an svelte component that I'd like to start the transition animation as it's being mounted even on first render.
Where exactly do I have to set intro: true?

Thanks!

**Comment by geoffrich** [comment]
Open issue with some workarounds in the comments here: https://github.com/sveltejs/kit/issues/2759

**Reply by ioucyf** [reply]
Thanks!

**Comment by ioucyf** [comment]
I have another question:

is it possible to revert back to svelte 3 until this is resolved? or is it going to break more things by doing so?

I'm on a completely empty sveltekit skeleton project. I'm starting fresh, but it comes with svelte 4.

**Reply by geoffrich** [reply]
Reverting back to Svelte 3 won't change anything about how the `intro` option works. What is breaking for you?

**Reply by ioucyf** [reply]
Transition animation on initial render when the component first mounts doesn't start. For Svelte 4, intro needs to be set to true during component creation in addition to adding the |global modifier. In Svelte 3, this wasn't an issue, and by default all transitions are set to |global unless |local flag is explicitly added.
